### ISSUE TITLE
[BUG] App does not start on Windows 11 Pro Version 22H2 Build 22621.2283

### ISSUE THREAD

**Issue by fubaWoW** [radj307/volume-control#101]
### Version

6.2.0

### Windows Version

Windows 11 Pro Version 22H2 Build 22621.2283

### Actual Behaviour

App does not start on Windows 11 Pro Version 22H2 Build 22621.2283

### Expected Behaviour

Guess what? ^^

### Reproduction Steps

1. Intalled the App 6.2.0 Revision 1 with ``VolumeControl-Installer.exe`` from https://github.com/radj307/volume-control/releases/tag/6.2.0-rev1
2. Tried to start the App with Desktop Link
3. Tried to start the App with Desktop Link as Administrator
4. Tried to start the App with ``VolumeControl.exe`` from the 
5. Tried to start the App with ``VolumeControl.exe`` as Administrator

no error, no message, nothing -> just does not start!

I do not use Windows Defender so there is no "Unblock"
I'm using Norton and it does not care anything about VolumeControl, means not even False-Positive!

there are also no logs, not in ``...\AppData\Local\Programs\VolumeControl`` (exe location) and also no directory like ``...\AppData\Local\radj307``

so...
why it does not start? ^^

**Comment by radj307** [comment]
Cannot replicate on Windows 11 Pro 22H2 Build 22621.1702 VM following the same steps (application opened normally on step 2).

Please try creating the `.../AppData/Local/radj307/VolumeControl` directory tree if it doesn't exist and try starting the program again. The program is supposed to create it if it doesn't exist, but you never know.

**Comment by radj307** [comment]
If the above steps don't work, try starting the app in a terminal and see if there is any output.

1. Open PowerShell or a command prompt window by R+Clicking on the start menu and selecting Windows PowerShell/Command Prompt
2. Navigate to volume control's installation directory.  
  `cd ~/AppData/Local/Programs/VolumeControl`
3. Run the application  
  `./VolumeControl.exe`
4. Post any messages it prints out here.

**Comment by fubaWoW** [comment]
```powershell
PowerShell 7.3.8
[C:\Users\fuba] # cd appdata\local
[C:\Users\fuba\AppData\Local] # cd .\Programs\
[C:\Users\fuba\AppData\Local\Programs] # cd .\VolumeControl\
[C:\Users\fuba\AppData\Local\Programs\VolumeControl] # .\VolumeControl.exe
[C:\Users\fuba\AppData\Local\Programs\VolumeControl] #
```
it outputs nothing

**Comment by radj307** [comment]
Do you have [.NET Core 6](https://dotnet.microsoft.com/en-us/download/dotnet/6.0) (or newer) installed? It would look something like this in the Add or Remove Programs window:  
![image](https://github.com/radj307/volume-control/assets/1927798/5d7e265d-bdf0-4403-8726-a9c26d013a24)  
The installer should have set it up automatically, and even when it's missing there's usually a message that appears to tell you it's not installed.  

The only other thing I can think of is to check the windows event logs (Start Menu -> Event Viewer). When the application fails to start you should be able to see an error with .NET Runtime as the Source.

**Comment by radj307** [comment]
If that doesn't work, b804b6274502123066d27e7a257c8cfd978d866e adds a wrapper around Main to hopefully catch and log any exceptions to STDERR (visible in terminal). Additionally, I added return codes to help with diagnosing the problem.

Will update shortly with a link to the build.

**Comment by fubaWoW** [comment]
Yes i have installed all .Net Runtimes from 3.5 up to 7.0.13 including 6.0.24 from update yesterday.

Also tried the new Version https://github.com/radj307/volume-control/releases/tag/6.3.0 with the same results and also no Console Message in Powershell...
I will test it soon on a VM because it seems very strange and my guess is Norton is Blocking it "silently" maybe?

i will report back soon!

**Comment by radj307** [comment]
Note: It shouldn't be necessary to have all of the different versions of the .NET Runtime installed; newer versions are backwards compatible. (that doesn't apply to .NET Framework or the SDKs, however)

Given that there still wasn't any output with 6.3.0, you're probably right about Norton being the cause. I haven't used it before so I wouldn't know for sure, but it's definitely possible for it to be silently preventing the application from starting. If it has an exclusion list that you can modify, try adding `VolumeControl.exe` and/or the installation directory to it before installing a whole virtual machine.

**Comment by fubaWoW** [comment]
Well...
I've now tested https://github.com/radj307/volume-control/releases/tag/6.3.0  it on a **FRESH** installed Windows 11 (see screenshot 1 for Build) on a VM with Windows Defender active.

Result is the same -> No start and no message at Powershell (see Scrennshot 2)

also tested Powershell Command:
``Unblock-File -Path "C:\Users\fuba\AppData\Local\Programs\VolumeControl\VolumeControl.exe"``
because there is not "Unblock" in my File-Properties! (see Screenshot 3)

my OS is German Language on both Main System and VM System but this hopefully can not be the problem ^^

i really don't know what is going wrong...

Screenshot 1
![image](https://github.com/radj307/volume-control/assets/41425685/f78edb27-efbb-459e-ae7b-830443983864)

Screenshot 2
![image](https://github.com/radj307/volume-control/assets/41425685/8cf4af4f-ab66-4a56-a349-42e58c74c660)

Screenshot 3
![image](https://github.com/radj307/volume-control/assets/41425685/444a262c-b5c5-4f7b-ac1d-12113afbc8b7)

**Comment by radj307** [comment]
The unblock checkbox only appears with the portable version AFAIK, so it's no surprise that it doesn't appear for you when using the installer.

I'm starting to wonder if the problem is related to the OS language being set to German, #107 is having a similar problem & one of the exception messages in their log is in German.

**Comment by radj307** [comment]
Confirmed - Previously working Volume Control installation stops working when the Windows 11 system language is set to the German (Germany) language pack.

**Comment by radj307** [comment]
Well this is really strange, after setting the LogFilter to 95 and changing the language pack *back* to German the issue is gone, and the application works again.

**Comment by fubaWoW** [comment]
ok...
now i've tried the 6.4.0 and it's still the same problem BUT just for the installer version.
the portable version works... but... why? 🤣

**Comment by radj307** [comment]
That's pretty strange. The version of `VolumeControl.exe` used by the installer is *slightly* different from the portable one, so maybe one of those differences is also a factor.

**Comment by radj307** [comment]
Could be related to the installer version being a 64-bit process.  

Now compiling installer version in x86 as of 2e4826c86acda8a72fca1c695481940f1e565def

**Comment by radj307** [comment]
Created version `6.4.0-dbg1` to test the 32-bit installer.  

Please download the [build artifact](https://github.com/radj307/volume-control/suites/17730372567/artifacts/1015167898) and see if the installer version works for you.

**Comment by fubaWoW** [comment]
very interesting...
both files in the package work, the portable exe as well as the installer version!
the installer now also create a log folder inside the folder ``AppData\Local\radj307\VolumeControl``

**Comment by radj307** [comment]
Good to hear! These changes will be included in the next release.  

That all but confirms that the issue was the 64-bit executable used by the installer, and also explains why you were unable to get any kind of output at all from the program.

I'm a bit confused as to why this was causing problems though, since AFAIK Windows 11 is 64-bit only. (This can be checked by running `wmic os get osarchitecture` in a terminal.)

**Comment by fubaWoW** [comment]
yeah im sure it is a 64-bit OS ^^
but why the f*** can this happen by only compile it in 64-bit?!
very strange and confusing 😂

but anyway it works now with the protable version for me and i think it will stay for now.

thanks

**Comment by radj307** [comment]
This issue has been fixed and is now available in [`v6.5.0`](https://github.com/radj307/volume-control/releases/tag/6.5.0).

**Comment by fubaWoW** [comment]
> This issue has been fixed and is now available in [`v6.5.0`](https://github.com/radj307/volume-control/releases/tag/6.5.0).

Thank you! 👍
